chapters/arrays.tex

```diff
@@ -580,6 +580,7 @@ \subsubsection{Reduction Expressions}\label{reduction-expressions}
 is a \firstuse{reduction expression}.
 The expressions in the iterators of a reduction expression shall be vector expressions.
 They are evaluated once for each reduction expression, and are evaluated in the scope immediately enclosing the reduction expression.
+If \lstinline!expression1! contains event-generating expressions the reduction expression shall be an evaluable expression.
```

Comment:
It doesn't make sense to require the entire reduction expression to be evaluable at the same time as it contains event-generating expressions.  I think this is what we mean:
```suggestion
If \lstinline!expression1! contains event-generating expressions, the expressions inside the iterators shall be evaluable.
```